src/Symfony/Component/ExpressionLanguage/ParserCache/ParserCacheAdapter.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\ExpressionLanguage\ParserCache;
+
+use Symfony\Component\Cache\Adapter\AbstractAdapter;
+use Psr\Cache\CacheItemInterface;
+use Symfony\Component\Cache\CacheItem;
+
+/**
+ * @author Alexandre GESLIN <alexandre@gesl.in>
+ *
+ * @internal
+ * 
+ * @deprecated ParserCacheAdapter is deprecated as of 3.2 and will be removed in 4.0
+ */
+class ParserCacheAdapter extends AbstractAdapter
+{
+    /**
+     * @var ParserCacheInterface
+     */
+    private $pool;
+
+    /**
+     * @var \Closure
+     */
+    private $createCacheItem;
+
+    public function __construct(ParserCacheInterface $pool)
+    {
+        $this->pool = $pool;
+        $defaultLifetime = 0;
+
+        $this->createCacheItem = \Closure::bind(
+            function ($key, $value, $isHit) use ($defaultLifetime) {
+                $item = new CacheItem();
+                $item->key = $key;
+                $item->value = $value;
+                $item->isHit = $isHit;
+                $item->defaultLifetime = $defaultLifetime;
+
+                return $item;
+            },
+            null,
+            CacheItem::class
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getItem($key)
```

Comment:
No problem ;) I will opt for an implementation of the PSR6 CacheItemPoolInterface directly. Less code, less dependencies, less work...